src/BSeries.jl

```diff
@@ -1754,10 +1754,13 @@ This function checks up to which order a Runge-Kutta method `rk` is
 energy-preserving for Hamiltonian problems.
 It requires a `max_order` so that it does not run forever if the order up to
 which the method is energy-preserving is too big or infinite.
+Keyword Arguments:
+- `atol::Float64=1e-14`: The absolute tolerance for energy preservation.
+- `rtol::Float64=1e-14`: The relative tolerance for energy preservation.
 
 See also [`is_energy_preserving`](@ref)
 """
-function energy_preserving_order(rk::RungeKuttaMethod, max_order)
+function energy_preserving_order(rk::RungeKuttaMethod, max_order; atol::Float64=1e-14, rtol::Float64=1e-14)
```

Comment:
We should not restrict the types so severely. Doing it like this means that we will not really be able to use other floating point precisions, for example.